iOS/DuckDuckGo/Refactoring/Common/OmniBarViewController.swift

```diff
@@ -0,0 +1,670 @@
+//
+//  OmniBarViewController.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+import PrivacyDashboard
+import Core
+
+class OmniBarViewController: UIViewController, OmniBar {
+
+    // MARK: - OmniBar conformance
+
+    // swiftlint:disable:next force_cast
+    var barView: any OmniBarView { view as! OmniBarView }
+
+    var isTextFieldEditing: Bool { false }
```

Comment:
Omid noticed that AI Chat isn't submitting text from the address bar and I think it's because this should delegate to the OmniBarView/textField? It doesn't appear to be overridden so always returns false.